src/core/lib/iomgr/resolve_address_posix.cc

```diff
@@ -77,6 +73,14 @@ static grpc_error* posix_blocking_resolve_address(
     port = default_port;
   }
 
+  if (host.compare("unix") == 0) {
+    return grpc_resolve_unix_domain_address(port.c_str(), addresses);
+  }
+
+  if (host.compare("unix-abstract") == 0) {
```

Comment:
I don't think this is the right place to do this. The code we have above this for "unix:" URIs is a really ugly hack that I've been meaning to get rid of. And even though I haven't yet gotten around to doing that, I don't want to repeat the existing mistake here.

The iomgr name resolution code is used both on the client and on the server, but the ability for it to understand "unix:" or "unix-abstract:" names is only needed on the server side, because the client side only uses this code when it already knows it's a DNS name -- if you specify a "unix:" or "unix-abstract:" URI on the client side, it selects a different resolver that does not call the iomgr name resolution code in the first place.

So instead of adding this hack here, we should instead move this check into the server code, which is the only place where it's actually needed.  I think the right place to do that is here:

https://github.com/grpc/grpc/blob/fa334f322fe3969b103c2292dfa79b6759e3ab72/src/core/ext/transport/chttp2/server/chttp2_server.cc#L281

That code should be changed to do something like this:

```
grpc_error* error = GRPC_ERROR_NONE;
if (absl::StartsWith(addr, "unix:")) {
  error = grpc_resolve_unix_domain_address(addr + sizeof("unix:"), &resolved);
} else if (absl::StartsWith(addr, "unix-abstract:")) {
  error = grpc_resolve_unix_abstract_domain_address(addr + sizeof("unix-abstract:"), &resolved);
} else {
  error = grpc_blocking_resolve_address(addr, "https", &resolved);
}
```

Then we can get rid of both of the special cases here.